File: Android/app/src/main/java/com/example/scheduleproject/Fragment/AccountFragment.java
package com.example.scheduleproject.Fragment;

import android.Manifest;
import android.app.Activity;
import android.content.ContentResolver;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.graphics.Bitmap;
import android.net.Uri;
import android.os.Bundle;

import androidx.core.app.ActivityCompat;
import androidx.fragment.app.Fragment;
import androidx.viewpager.widget.ViewPager;

import android.provider.MediaStore;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import android.widget.Toast;

import com.bumptech.glide.Glide;
import com.example.scheduleproject.R;
import com.google.android.material.badge.BadgeDrawable;
import com.gun0912.tedpermission.PermissionListener;
import com.gun0912.tedpermission.TedPermission;

import java.io.IOException;
import java.util.BitSet;
import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;
import gun0912.tedbottompicker.TedBottomPicker;
import gun0912.tedbottompicker.TedBottomSheetDialogFragment;

import static android.app.Activity.RESULT_OK;
import static com.bumptech.glide.load.resource.drawable.DrawableTransitionOptions.withCrossFade;

public class AccountFragment extends Fragment {
    View view;
    private CircleImageView btnAdd, img;
    TextView txtmail;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        view = inflater.inflate(R.layout.fragment_account, container, false);
        btnAdd = view.findViewById(R.id.img_plus);
        img = view.findViewById(R.id.img_profile);
        txtmail = view.findViewById(R.id.txtEmail);

        btnAdd.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (ActivityCompat.checkSelfPermission(getActivity(),
                        Manifest.permission.READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
                    requestPermissions(
                            new String[]{Manifest.permission.READ_EXTERNAL_STORAGE},
                            2000);
                } else {
                    Intent cameraIntent = new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
                    startGallery();
                }
            }
        });

        txtmail.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

            }
        });
        return view;
    }

    private void startGallery() {
        Intent cameraIntent = new Intent(Intent.ACTION_GET_CONTENT);
        cameraIntent.setType("image/*");
        if (cameraIntent.resolveActivity(getActivity().getPackageManager()) != null) {
            startActivityForResult(cameraIntent, 1000);
        }
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        //super method removed
        if (resultCode == RESULT_OK) {
            if (requestCode == 1000) {
                Uri returnUri = data.getData();
                Bitmap bitmapImage = null;
                try {
                    bitmapImage = MediaStore.Images.Media.getBitmap(getActivity().getContentResolver(), returnUri);
                } catch (IOException e) {
                    e.printStackTrace();
                }
                img.setImageBitmap(bitmapImage);

                Glide.with(this)
                        .load(returnUri)
                        .override(1280, 1280)
                        .centerCrop()
                        .transition(withCrossFade())
                        .into(img);
            }
        }
    }
}

File: Android/app/src/main/java/com/g5/hapiappdemo/fragments/AccountFragment.kt
package com.g5.hapiappdemo.fragments

import android.app.Activity.RESULT_OK
import android.app.ProgressDialog
import android.content.Intent
import android.database.Cursor
import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.net.Uri
import android.os.Bundle
import android.provider.MediaStore
import android.util.Base64
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import com.g5.hapiappdemo.PreferenceConstants
import com.g5.hapiappdemo.R
import com.g5.hapiappdemo.api.ApiClient
import com.g5.hapiappdemo.auth.set
import com.g5.hapiappdemo.databinding.FragmentAccountBinding
import com.g5.hapiappdemo.extensions.PreferenceHelper
import com.squareup.picasso.Picasso
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers
import java.io.ByteArrayOutputStream
import java.net.URLEncoder
import java.nio.charset.StandardCharsets


class AccountFragment : Fragment() {

    lateinit var binding: FragmentAccountBinding
    var URL = "https://notevn.com/file/vi_internal"
    private var disposable: Disposable? = null
    private var pd: ProgressDialog? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(false)
    }

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = DataBindingUtil.inflate(
            inflater,
            R.layout.fragment_account, container, false
        )
        return binding.root
    }

    private fun clickpic() {
        val i = Intent(
            Intent.ACTION_PICK,
            MediaStore.Images.Media.EXTERNAL_CONTENT_URI
        )
        startActivityForResult(i, 100)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == 100 && resultCode == RESULT_OK && null != data) {
            val selectedImage: Uri? = data.data
            val filePathColumn = arrayOf(MediaStore.Images.Media.DATA)
            val cursor: Cursor =
                requireContext().contentResolver?.query(
                    selectedImage!!,
                    filePathColumn, null, null, null
                )!!
            cursor!!.moveToFirst()
            val columnIndex: Int = cursor!!.getColumnIndex(filePathColumn[0])
            val picturePath: String = cursor!!.getString(columnIndex)
            cursor!!.close()

            this.upload(picturePath)
        }
    }

    private fun upload(picturePath: String) {
        pd!!.setCancelable(false)
        pd!!.setMessage(resources.getString(R.string.image_uploading_txt))
        pd!!.show()

        val bm = BitmapFactory.decodeFile(picturePath)
        val bao = ByteArrayOutputStream()
        bm.compress(Bitmap.CompressFormat.JPEG, 90, bao)
        val ba: ByteArray = bao.toByteArray()
        val base64 = Base64.encodeToString(ba, Base64.DEFAULT)

        if (ApiClient.getInstance(requireContext()).isReady()) {
            disposable = ApiClient.getInstance(requireActivity()).uploadAvatar(base64)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(
                    { result ->
                        val bitmap = BitmapFactory.decodeFile(picturePath)
                        val prefs = PreferenceHelper.securePrefs(requireContext())

                        binding.profileImage.setImageBitmap(bitmap)
                        prefs[PreferenceConstants.avatar] = "https://cdn.notevn.com/${result.file_name}${result.type}"
                        pd!!.hide()
                        pd!!.dismiss()
                    },
                    { _ ->
                        pd!!.hide();
                        pd!!.dismiss();
                        Toast.makeText(
                            requireContext(),
                            resources.getString(R.string.avatar_upload_error),
                            Toast.LENGTH_SHORT
                        ).show()
                    }
                )
        }
    }

    private fun initStudentData() {
        val uData = ApiClient.getInstance(requireActivity().applicationContext)

        binding.studentName.text = uData.getStudentName()
        binding.studentId.text = uData.getStudentID()
        binding.studentEmail.text = uData.getStudentEmail()
        binding.studentSdt1.text = uData.getStudentP1()
        binding.studentSdt2.text = uData.getStudentP2()
        Picasso.with(requireContext()).load(uData.getAvatar()).into(binding.profileImage)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        pd = ProgressDialog(requireContext(), R.style.Theme_AppCompat_Light_Dialog_Alert)

        this.initStudentData()
        binding.profileImage.setOnClickListener {
            clickpic()
        }
    }

    companion object {
        fun newInstance(): AccountFragment {
            return AccountFragment()
        }
    }
}
